gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -172,6 +196,10 @@ private void awaitAllOutstandingBatches() throws InterruptedException {
   public void close() throws InterruptedException {
     isClosed = true;
     flush();
+
+    if (!scheduledFuture.isDone()) {
```

Comment:
I think it would be better to add Preconditions.checkState(!isClosed) and unconditionally cancel the future

Reply:
I think `Preconditions.checkState(!isClosed)` inside `close()` would not be best, As if a user accidentally or intently calls `close()` more than once then it would throw `IllegalStateException`.

Please let me know if we should add some other check/log for `isClosed` flag.

Reply:
In that case I'd prefer `if(isClosed) return;`